Source/WebCore/dom/Document.cpp

```diff
@@ -3170,6 +3170,14 @@ bool Document::updateLayoutIfDimensionsOutOfDate(Element& element, OptionSet<Dim
         // Check our containing block chain. If anything in the chain needs a layout, then require a full layout.
         for (CheckedPtr currentRenderer = renderer; currentRenderer && !currentRenderer->isRenderView(); currentRenderer = currentRenderer->container()) {
 
+            if (currentRenderer->isSVGRenderer()) {
+                if (currentRenderer->selfNeedsLayout() || currentRenderer->needsPositionedMovementLayout()) {
+                    requireFullLayout = true;
+                    break;
+                }
+                continue;
```

Comment:
Do we need to check
```
    if (currentRenderer == frameView->layoutContext().subtreeLayoutRoot())
        break;
```
 like what we do at the end of this for-loop? This `continue` will skip this check.

Reply:
If `currentRenderer->isSVGRenderer()` but it corresponds to a `<use>` element and the position of one of the elements in the shadow tree was changed, will that set `requireFullLayout` eventually or not?

Reply:
The above refactoring means that we now hit the `subtreeLayoutRoot` check for all types. This wasn't strictly necessary, but can be a nice early exit.

Changing content inside the `<use>` should set layout flags on the ancestor that stops this fast path from being taken.